ome/omero-web
# Filter by map anns webclient

As requested in various workshops, allow filtering of images by Map Annotations (without needing to install OMERO.parade).
This also recognises when all the values for a chosen Key are numbers, and then lets you do numerical filtering.

![Screen Shot 2019-10-03 at 10 24 03](https://user-images.githubusercontent.com/900055/66115064-0153cd80-e5c8-11e9-8685-af4363046372.png)

To test:
 - Pick a Dataset of Images with a good selection of Map Annotations (all namespaces are used), including some 'numerical' values.
 - Choose to filter by 'Key-Values'.
 - Pick a Key - images without that key will be immediately filtered out
 - If you've picked a Key where all the values are numbers, you will see a ```>``` chooser and the min-max range will be shown as a placeholder.
 - Otherwise, you'll get a regular text filter.
 - Typing in the filter field will start filtering.
 - Additional Map-Annotation filters can be added, also with other filters.
 - Clicking the X will remove the filter and un-filter the images.

cc @pwalczysko
